homeassistant/components/tuya/vacuum.py

```diff
@@ -94,7 +95,13 @@ def __init__(self, device: TuyaDevice, device_manager: TuyaDeviceManager) -> Non
         if self.find_dpcode(DPCode.PAUSE, prefer_function=True):
             self._supported_features |= SUPPORT_PAUSE
 
-        if self.find_dpcode(DPCode.SWITCH_CHARGE, prefer_function=True):
+        if (
+            self.find_dpcode(DPCode.SWITCH_CHARGE, prefer_function=True)
+            or TUYA_MODE_RETURN_HOME
+            in EnumTypeData.from_json(
+                DPCode.MODE, device.status_range.get(DPCode.MODE).values
+            ).range
```

Comment:
Please don't instantiate EnumTypeData manually, instead use `find_dpcode` to do that for you.

Reply:
Using `find_dpcode`, tests do not pass due to error `Item "None" of "Optional[EnumTypeData]" has no attribute "range"`, despite having the range attribute. It works fine locally.

Can you help me indicating the way to solve the test error?

Code:
````
        if (
            self.find_dpcode(DPCode.SWITCH_CHARGE, prefer_function=True)
            or TUYA_MODE_RETURN_HOME
            in self.find_dpcode(DPCode.MODE, dptype=DPType.ENUM).range
        ):
            self._supported_features |= SUPPORT_RETURN_HOME
````

Error:
````
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

homeassistant/components/tuya/vacuum.py:101: error: Item "None" of "Optional[EnumTypeData]" has no attribute "range"  [union-attr]
Found 1 error in 1 file (checked 1 source file)
````

Reply:
> Item "None" of "Optional[EnumTypeData]" has no attribute "range"

`find_dpcode` can return `None` in case the dpcode isn't found. This needs to be guarded.

Reply:
Can you please rate the changes?